refine processor log output when a changefeed is stopped

## Feature Request

**Is your feature request related to a problem? Please describe:**
<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->
- start a cdc capture
- start a changefeed, having several tables to be replicated in upstream
- stop the changefeed via `curl -X POST -d "admin-job=3&cf-id=xxx`

**Describe the feature you'd like:**
<!-- A clear and concise description of what you want to happen. -->

there exists some error logs

```
[2020/05/08 12:39:26.867 +08:00] [ERROR] [client.go:920] ["failed to receive from stream"] [addr=127.0.0.1:20160] [storeID=1] [error="rpc error: code = Canceled desc = context canceled"]
[2020/05/08 12:39:26.867 +08:00] [INFO] [client.go:890] ["stream to store closed"] [addr=127.0.0.1:20160] [storeID=1]
[2020/05/08 12:39:31.865 +08:00] [ERROR] [processor.go:279] ["failed to update info"] [error="failed to update info: changefeed: 6f0fa8f6-8700-436c-a410-a38ed03e6459, capture: ce9312d0-e2e6-4684-9d7e-ae58f552d999: task status not exists"] [errorVerbose="task status not exists\nchangefeed: 6f0fa8f6-8700-436c-a410-a38ed03e6459, capture: ce9312d0-e2e6-4684-9d7e-ae58f552d999\ngithub.com/pingcap/ticdc/cdc/kv.CDCEtcdClient.GetTaskStatus\n\tgithub.com/pingcap/ticdc@/cdc/kv/etcd.go:307\ngithub.com/pingcap/ticdc/cdc/roles/storage.(*ProcessorTsEtcdRWriter).UpdateInfo\n\tgithub.com/pingcap/ticdc@/cdc/roles/storage/etcd.go:93\ngithub.com/pingcap/ticdc/cdc.(*processor).updateInfo\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:373\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker.func1.1\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:259\ngithub.com/pingcap/ticdc/pkg/retry.Run.func1\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:31\ngithub.com/cenkalti/backoff.RetryNotify\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37\ngithub.com/cenkalti/backoff.Retry\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24\ngithub.com/pingcap/ticdc/pkg/retry.Run\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:30\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker.func1\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:258\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker.func2\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:277\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:329\ngithub.com/pingcap/ticdc/cdc.(*processor).Run.func1\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:188\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20190423024810-112230192c58/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357\nfailed to update info"]
[2020/05/08 12:39:31.866 +08:00] [INFO] [processor.go:282] ["Local resolved worker exited"]
[2020/05/08 12:39:31.866 +08:00] [INFO] [processor.go:582] ["syncResolved stopped"]
[2020/05/08 12:39:31.866 +08:00] [ERROR] [client.go:920] ["failed to receive from stream"] [addr=127.0.0.1:20160] [storeID=1] [error="rpc error: code = Canceled desc = context canceled"]
[2020/05/08 12:39:31.866 +08:00] [INFO] [client.go:890] ["stream to store closed"] [addr=127.0.0.1:20160] [storeID=1]
[2020/05/08 12:39:31.866 +08:00] [ERROR] [client.go:920] ["failed to receive from stream"] [addr=127.0.0.1:20160] [storeID=1] [error="rpc error: code = Canceled desc = context canceled"]
[2020/05/08 12:39:31.866 +08:00] [INFO] [client.go:890] ["stream to store closed"] [addr=127.0.0.1:20160] [storeID=1]
[2020/05/08 12:39:32.072 +08:00] [INFO] [processor.go:515] ["Global resolved worker exited"]
[2020/05/08 12:39:33.869 +08:00] [ERROR] [processor.go:775] ["error on running processor"] [captureid=ce9312d0-e2e6-4684-9d7e-ae58f552d999] [changefeedid=6f0fa8f6-8700-436c-a410-a38ed03e6459] [processorid=575cd693-c81a-43bd-a7cf-8a90ad90d8fa] [error="failed to update info: changefeed: 6f0fa8f6-8700-436c-a410-a38ed03e6459, capture: ce9312d0-e2e6-4684-9d7e-ae58f552d999: task status not exists"] [errorVerbose="task status not exists\nchangefeed: 6f0fa8f6-8700-436c-a410-a38ed03e6459, capture: ce9312d0-e2e6-4684-9d7e-ae58f552d999\ngithub.com/pingcap/ticdc/cdc/kv.CDCEtcdClient.GetTaskStatus\n\tgithub.com/pingcap/ticdc@/cdc/kv/etcd.go:307\ngithub.com/pingcap/ticdc/cdc/roles/storage.(*ProcessorTsEtcdRWriter).UpdateInfo\n\tgithub.com/pingcap/ticdc@/cdc/roles/storage/etcd.go:93\ngithub.com/pingcap/ticdc/cdc.(*processor).updateInfo\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:373\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker.func1.1\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:259\ngithub.com/pingcap/ticdc/pkg/retry.Run.func1\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:31\ngithub.com/cenkalti/backoff.RetryNotify\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37\ngithub.com/cenkalti/backoff.Retry\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24\ngithub.com/pingcap/ticdc/pkg/retry.Run\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:30\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker.func1\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:258\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:327\ngithub.com/pingcap/ticdc/cdc.(*processor).Run.func1\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:188\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20190423024810-112230192c58/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357\nfailed to update info"]
```

--- a/cdc/processor.go
+++ b/cdc/processor.go
@@ -69,6 +69,7 @@ type processor struct {
 	changefeedID string
 	changefeed   model.ChangeFeedInfo
 	limitter     *puller.BlurResourceLimitter
+	stopped      int32
 
 	pdCli     pd.Client
 	kvStorage tidbkv.Storage
@@ -261,8 +262,10 @@ func (p *processor) positionWorker(ctx context.Context) error {
 		t0Update := time.Now()
 		err := retry.Run(500*time.Millisecond, 3, func() error {
 			inErr := p.updateInfo(ctx)
-			if errors.Cause(inErr) == model.ErrAdminStopProcessor {
-				return backoff.Permanent(inErr)
+			if inErr != nil {
+				if p.isStopped() || errors.Cause(inErr) == model.ErrAdminStopProcessor {
+					return backoff.Permanent(errors.Trace(model.ErrAdminStopProcessor))
+				}
 			}
 			return inErr
 		})
@@ -278,9 +281,11 @@ func (p *processor) positionWorker(ctx context.Context) error {
 		resolveTsTick.Stop()
 		checkpointTsTick.Stop()
 
-		err := updateInfo()
-		if err != nil && errors.Cause(err) != context.Canceled {
-			log.Error("failed to update info", zap.Error(err))
+		if !p.isStopped() {
+			err := updateInfo()
+			if err != nil && errors.Cause(err) != context.Canceled {
+				log.Warn("failed to update info before exit", zap.Error(err))
+			}
 		}
 
 		log.Info("Local resolved worker exited")
@@ -747,16 +752,22 @@ func (p *processor) addTable(ctx context.Context, tableID int64, startTs uint64)
 }
 
 func (p *processor) stop(ctx context.Context) error {
+	log.Info("stop processor", zap.String("id", p.id), zap.String("capture", p.captureID), zap.String("changefeed", p.changefeedID))
 	p.tablesMu.Lock()
 	for _, tbl := range p.tables {
 		tbl.cancel()
 	}
 	p.tablesMu.Unlock()
+	atomic.StoreInt32(&p.stopped, 1)
 
 	if err := p.etcdCli.DeleteTaskPosition(ctx, p.changefeedID, p.captureID); err != nil {
 		return err
 	}
 	return p.etcdCli.DeleteTaskStatus(ctx, p.changefeedID, p.captureID)
+}
+
+func (p *processor) isStopped() bool {
+	return atomic.LoadInt32(&p.stopped) == 1
 }
 
 // runProcessor creates a new processor then starts it.
@@ -798,16 +809,15 @@ func runProcessor(
 
 	go func() {
 		err := <-errCh
-		if errors.Cause(err) != context.Canceled {
+		cause := errors.Cause(err)
+		if cause != nil && cause != context.Canceled && cause != model.ErrAdminStopProcessor {
+			processorErrorCounter.WithLabelValues(changefeedID, captureID).Inc()
 			log.Error("error on running processor",
 				zap.String("captureid", captureID),
 				zap.String("changefeedid", changefeedID),
 				zap.String("processorid", processor.id),
 				zap.Error(err))
 		} else {
-			if err != nil {
-				processorErrorCounter.WithLabelValues(changefeedID, captureID).Inc()
-			}
 			log.Info("processor exited",
 				zap.String("captureid", captureID),
 				zap.String("changefeedid", changefeedID),

